ikarus/finiteelements/mechanics/materials/vanishingstress.hh

```diff
@@ -253,10 +253,13 @@ private:
         E(indexPair[0], indexPair[1]) += ecomps(ri++);
       }
     };
+
+    int minIter = std::is_same_v<ScalarType, double> ? 0 : 1;
```

Comment:
Maybe we can have a concept IsAutoDiff or sth like that, maybe we want to use the material with long double or so

Reply:
I thought about that, but `VanishingStress` only knows if AutoDiff is used by the `ScalarType` or if the `rebind` function is called? So how exactly would this concept be that could perhaps be applicable also somewhere else? May be using `bool isAutoDiff = not(std::is_integral_v<ScalarType> and is_floating_point_v<ScalarType>)`?

Reply:
Also, I think everywhere, this is the only check that is done to differentiate from `AutoDiff`. See also [here](https://github.com/ikarus-project/ikarus/blob/d05187eb1cae63f6a2f205baa70776701fa9f40c/ikarus/finiteelements/mechanics/nonlinearelastic.hh#L155), for example. If support for `long double`, etc. is needed, probably refactoring has to be done at several other places?

Reply:
Can you check if this is helpful? 
https://github.com/autodiff/autodiff/blob/cb2d6e8b9fa98b7b8cf3ffaefa231943c56361ae/autodiff/forward/dual/dual.hpp#L380C1-L381C61

Otherwise I wouls go for `is_floating_point_v<ScalarType>`

Reply:
I will use `std::is_floating_point_v<ScalarType>` for now. I will try to draft up a concept using [this](https://github.com/autodiff/autodiff/blob/cb2d6e8b9fa98b7b8cf3ffaefa231943c56361ae/autodiff/forward/dual/dual.hpp#L380C1-L381C61) separately and incorporate it everywhere else in Ikarus